src/grz_cli/transfer.py

```diff
@@ -40,3 +40,25 @@ def init_s3_client(config: ConfigModel) -> boto3.session.Session.client:
     )
 
     return s3_client
+
+
+def init_s3_resource(config: ConfigModel) -> boto3.resources.base.ServiceResource:
```

Comment:
Should everything eventually migrate from the boto3 Client API to the Resource API?

Reply:
Right now it doesn't make a big difference (neither in functionality nor in lines of code / style). Internally, both use S3Transfer.
However, if we want to implement continuing multipart uploads in the future, we have to use the boto3 client directly.